examples/SplitKernels/main.cu

```diff
@@ -175,8 +175,13 @@ void runGPU(int numParticles, double energy, int batch, const int *MCIndex_host,
   COPCORE_CUDA_CHECK(cudaMemcpy(MCIndex_dev, MCIndex_host, sizeof(int) * numVolumes, cudaMemcpyHostToDevice));
   COPCORE_CUDA_CHECK(cudaMemcpyToSymbol(MCIndex, &MCIndex_dev, sizeof(int *)));
 
+  cudaDeviceProp deviceProp;
+  cudaGetDeviceProperties(&deviceProp, 0);
+
   // Capacity of the different containers aka the maximum number of particles.
-  constexpr int Capacity = 1024 * 1024;
+  // Use 1/5 of GPU memory for each of e+/e-/gammas, leaving 2/5 for the rest.
+  const size_t Capacity = (deviceProp.totalGlobalMem / sizeof(Track) / 5);
+  const int MaxBlocks = deviceProp.maxGridSize[0];
```

Comment:
For all the reasons that we discussed many times by now and I keep repeating (last time in https://github.com/apt-sim/AdePT/pull/214), I'm against doing this.

Reply:
I disagree. I think it's reasonable to support as many tracks as the GPU can handle. Like this, you can test various batch sizes via the command line without having to recompile, and you push the out-of-memory crash as far out as possible.
It doesn't mean that you will always run with the maximum batch size.

Reply:
If that's the issue, we can just pin the default size to 100, or another number that you can choose here.

Reply:
I don't think these batch sizes are unrealistic in the future. They may require changes on the Geant4 workflow though, but that should not matter now. This new example is a playground for R&D on we are trying to get the most out of the GPU, knowing that our changes may not fit the current Geant4 workflow.

Reply:
Let me once again explain what I'm after: I want a realistic scenario be bested by default, and in a way that is comparable to existing examples.